pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -124,3 +147,100 @@ func Redis(apimanager *appsv1alpha1.APIManager, client client.Client) (*componen
 	}
 	return component.NewRedis(opts), nil
 }
+
+func (r *RedisReconciler) deleteSystemRedisSecretNamespaceAttribute() (reconcile.Result, error) {
+	redis, err := Redis(r.apiManager, r.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	desired := redis.SystemRedisSecret()
```

Comment:
In general, I think it's better to fetch secret directly. We know the secret name (system-redis) and namespace (from apim)

Reply:
Done. Thank you